src/macaron/slsa_analyzer/git_url.py

```diff
@@ -177,7 +176,12 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "")
             logger.info("Cannot find commit %s on branch %s. Please check the configuration.", digest, res_branch)
             return False
 
-    logger.info("Successfully checked out commit %s.", head_commit)
+    head_commit: Commit = git_obj.repo.head.commit
+    if not head_commit:
```

Comment:
Fixed in https://github.com/oracle-samples/macaron/pull/125/commits/dc986c6b6888c740e1653634c71449ac4bcfdcc7

Reply:
But still we need to check if `git_obj.repo.head.commit` is not `None` [here](https://github.com/oracle-samples/macaron/blob/117-macaron-pull-the-latest-when-analyzing-a-local-cloned-repository/src/macaron/slsa_analyzer/git_url.py#L159). Otherwise it might throw an exception.

Reply:
If `current_head` is None here, should we return False early or should we proceed to try to check out the specific commit by the user :thinking: . The Exceptions for checking if `digest` exists or whether we can checkout `digest`  (after [here](https://github.com/oracle-samples/macaron/blob/117-macaron-pull-the-latest-when-analyzing-a-local-cloned-repository/src/macaron/slsa_analyzer/git_url.py#L159)) should be caught and return False.